### PR TITLE
Doc: Add topic and expand info for in-memory queue

### PR DESCRIPTION
**Pull request by karenzone**
## Release notes
[rn:skip] 

## What does this PR do?

Adds topic with considerations and config info related to in-memory queue. This approach facilitates easier comparison and makes info on par with PQ info. 

## Why is it important/What is the impact to the user?

Users need more info on sizing the internal memory queue and need to know that settings are per pipeline.

Related: #13232 

**PREVIEW:** https://logstash_13246.docs-preview.app.elstc.co/guide/en/logstash/master/memory-queue.html

### CODE REVIEW

**Comment by karenzone** [comment]
@robbavey  Let's be sure we're on the same page with this approach, please.

File: docs/static/resiliency.asciidoc [hunk 1]
```diff
@@ -1,6 +1,8 @@
 [[resiliency]]
 == Data resiliency
```
Comment (karenzone) [inline docs/static/resiliency.asciidoc:2]: @robbavey What do you think about this heading? I've expanded the info to include (default) mem queue? Something like "Queues and data resiliency" maybe?  Other ideas are welcomed.

File: docs/static/mem-queue.asciidoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+[[backpressure-mem-queue]]
+==== Back pressure
+
+When the queue is full, Logstash puts back pressure on the inputs to stall data
+flowing into Logstash. 
+This mechanism helps Logstash control the rate of data flow at the input stage
+without overwhelming outputs like Elasticsearch.
+
+ToDo: Is the next paragraph accurate for MQ?
```
[the remark this answers is not in the document]
Comment (karenzone) [reply docs/static/mem-queue.asciidoc:41]: As that PR is WIP, does it make sense to delete this part: 
> For example, when the
> <<plugins-inputs-beats,beats input>> encounters back pressure, it no longer
> accepts new connections.
> It waits until the queue has space to accept more events. 
> After the filter and output stages finish processing existing
> events in the queue and ACKs them, Logstash automatically starts accepting new
> events.

If not, what's your recommendation on making this right?

[the remark this answers is not in the document]
Comment (karenzone) [reply docs/static/mem-queue.asciidoc:41]: ToDo: @karenzone to open an issue to detail how different inputs handle backpressure.

File: docs/static/mem-queue.asciidoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+* Can lose data in abnormal termination
+* Don't do well handling sudden bursts of data, where extra capacity in needed for {ls} to catch up
+
+TIP: Consider using <<persistent-queues,persistent queues>> to avoid these limitations. 
+
+[[sizing-mem-queue]]
+==== Memory queue size
+
+Memory queue size is not configured directly.
+It is defined by the number of events, which can vary greatly depending on the event payload.
```
Comment (robbavey) [inline docs/static/mem-queue.asciidoc:29]: Maybe refer to the size of event here?
```suggestion
It is defined by the number of events, the size of which can vary greatly depending on the event payload. 
```

**Review by robbavey — COMMENTED** [review]
One small suggestion, but I think we're _really_ close

**Review by robbavey — APPROVED** [review]
LGTM

File: docs/static/mem-queue.asciidoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+* Don't do well handling sudden bursts of data, where extra capacity in needed for {ls} to catch up
+* Not a good choice for data you can't afford to lose
+
+TIP: Consider using <<persistent-queues,persistent queues>> to avoid these limitations. 
+
+[[sizing-mem-queue]]
+==== Memory queue size
+
+Memory queue size is not configured directly.
+Multiply the `pipeline.batch.size` and `pipeline.workers` values to get the size of the memory queue.
```
[the remark this answers is not in the document]
Comment (karenzone) [reply docs/static/mem-queue.asciidoc]: > Pipeline->pipeline has its own complexity, but I think that might be something we could talk about in a separate PR. Maybe in an advanced section, or pipeline->pipeline considerations?

New issue to track: #13275

**Pull request merged**